### PR TITLE
[Merged by Bors] - ET-3602 cleanup docker image

### PR DESCRIPTION
**Pull request by janpetschexain**
**Reference**

-  [ET-3602]

**Summary**

- cleanup env vars
- cleanup docker image

[ET-3602]: https://xainag.atlassian.net/browse/ET-3602?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/docker/Dockerfile.ci-image [hunk 1]
```diff
@@ -19,87 +12,8 @@ RUN apt-get update && apt-get install -y \
   rsync \
   && rm -rf /var/lib/apt/lists/*
```
Comment (janpetschexain) [inline .github/docker/Dockerfile.ci-image:13]: do we still need lines 10-13?

Comment (acrrd) [reply .github/docker/Dockerfile.ci-image:13]: clang was needed for bindgen, rsync to for the release ci.
We need to add `wget curl build-essential`, they are installed in another pass that is removed.

**Comment by janpetschexain** [comment]
checked that it builds: [link](https://github.com/xaynetwork/xayn_discovery_engine/actions/runs/3470839825/jobs/5799546620)

**Review by acrrd — APPROVED** [review]

**Comment by janpetschexain** [comment]
bors merge

**Pull request closed without merging**